alphalens/performance.py

```diff
@@ -247,7 +247,11 @@ def quantize_factor(factor, quantiles=5, by_group=False):
     """
 
     def quantile_calc(x, quantiles):
-        return pd.qcut(x, quantiles, labels=False) + 1
+        try:
+            return pd.qcut(x, quantiles, labels=False) + 1
+        except ValueError:
+            # This is supposed to fix "ValueError: Bin edges must be unique"
+            return pd.qcut(x.rank(method='first'), quantiles, labels=False) + 1
```

Comment:
@jameschristopher I understand that and it is the reason why I explicitly stated this behaviour. Also I actually opened this PR mostly to listen to your opinions and have a discussion on this exception. I don't really expect to have this PR merged if the pros are less than the cons.

Imagine we have a factor whose values are 50% identical. If we run the tear sheet with a quantiles value different than 2 we'll get a "ValueError: Bin edges must be unique"  excepion because the identical values would split in more than one quantile and this would lead to bid edges with the same value. 

With this PR we would avoid the exception but the quantiles that contain (fully or partially) the identical values are arbitrary, as we could swap securities with identical factor values between quantiles. The consequence of having generated arbitrary quantiles is that we can only trust the alphalens statistics calculated for the quantiles that do not contain the identical values (the quantiles that are not arbitrary) and we have to look at the quantiles with identical values as just one of the multiple possibilities the identical values can be split into.

So, I can see the limits of this PR but there are the cases where the identical values are only a very small percentage of the total amount (still an exception can be raised if the a bin edge falls in the middle of the identical value group) and in that case the skew would be small.